addons/hr_homeworking/models/res_partner.py

```diff
@@ -7,19 +7,14 @@ class ResPartner(models.Model):
 
     def _compute_im_status(self):
         super()._compute_im_status()
-        users = self.env["res.users"].search([
-            ('partner_id.id', 'in', self.ids)])
-        for user in users:
-            dayfield = self.env['hr.employee']._get_current_day_location_field()
-            location_type = user[dayfield].location_type
+        dayfield = self.env['hr.employee']._get_current_day_location_field()
+        for partner in self:
+            location_type = self.employee_id[dayfield].location_type
```

Comment:
here you can do `self.employee_ids[0]` no?

Reply:
I need to filter by company first and after to take the first :thinking: and I will use it two times so I don't know if it's better or not.

Reply:
for that we don't need extra field ^^